Modules/xASL_module_ASL.m

```diff
@@ -421,21 +459,25 @@
 if ~x.mutex.HasState(StateName{iState}) && x.mutex.HasState(StateName{iState-4})
 
     fprintf('%s\n','Quantifying ASL:   ');
-    if isfield(x.Q, 'bUseBasilQuantification') && x.Q.bUseBasilQuantification
-        % Quantification in native space:
+    if isfield(x.Q, 'bUseBasilQuantification') && x.Q.bUseBasilQuantification % if BASIL quantification will be performed, only native space analysis is possible
+        % Quantification in native space only:
         nVolumes = size(xASL_io_Nifti2Im(x.P.Path_ASL4D), 4);
         
         if nVolumes>1
-            xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient, 1);
+            x = xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient, 1); 
         else
-            xASL_wrp_Quantify(x, x.P.Path_PWI, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient, 1);
-        end        
-    end
-    
+            x = xASL_wrp_Quantify(x, x.P.Path_PWI, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient, 1);
+        end   
+    else
```

Comment:
So until now, when we had BASIL set for single-PLD, it outputted both CBF.nii and CBF_BASIL.nii.
I am not against this - if you decide to go for BASIL, you go for BASIL. But - we need to make changes in the TestDataSet - because currently, it is running 1-2 BASIL quantification, but really evaluating the non-BASIL output.

So I propose to do the following:
1. Keep here your change of running either normal or BASIL quantification.
2. But make sure that it always outputs only the BASIL output as CBF.nii
3. Revert the BASIL option in TestDataSet, or make sure that BASIL and no-BASIL CBF are equal, or we will have issues with it later.

Reply:
Yes I thought so as well. Makes the code simpler as well, as you don't need if `bUseBasilQuantification` in subsequent wrappers. Would be good then to state at several locations a clear message (not a warning) that Basil quantification is used.

Reply:
Okay I will change all BASIL output to CBF.nii, instead of the current CBF_BASIL.nii. Will add the warnings as well.